scrapy_streaming/communication/map.py

```diff
@@ -1,24 +1,31 @@
-import json
+import base64
+
+import simplejson as json
 
 from scrapy.utils.python import to_unicode, to_native_str
 
-from scrapy_streaming.communication import wrappers
-from scrapy_streaming.utils import MessageError
+from scrapy_streaming.communication import validators
+from scrapy_streaming.utils import MessageError, extract_instance_fields
 
 
 class CommunicationMap(object):
     """
-    Helper class to create the json messages
+    Helper class to create and receive json messages
     """
 
     mapping = {
-        'spider': wrappers.SpiderMessage,
-        'request': wrappers.RequestMessage,
-        'log': wrappers.LogMessage
+        'spider': validators.SpiderMessage,
+        'request': validators.RequestMessage,
+        'form_request': validators.FormRequestMessage,
```

Comment:
let's improve this name, as a scrapy `FormRequest` is just a `POST` request, while I think you are trying to do here a `from_response_request`